src/protagonist/API/Features/NamedQueries/NamedQueriesController.cs

```diff
@@ -0,0 +1,166 @@
+using API.Infrastructure;
+using API.Settings;
+using DLCS.Core.Collections;
+using DLCS.HydraModel;
+using DLCS.Web.Auth;
+using MediatR;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+
+namespace API.Features.NamedQueries;
+
+/// <summary>
+/// DLCS REST API Operations for Named Queries
+/// </summary>
+[Route("/customers/{customerId}/namedQueries")]
+[ApiController]
+public class NamedQueriesController : HydraController
+{
+    public NamedQueriesController(
+        IMediator mediator,
+        IOptions<ApiSettings> options) : base(options.Value, mediator)
+    {
+    }
+    
+    /// <summary>
+    /// Get a list of all available Named Queries, either global or owned by the user
+    /// </summary>
+    /// <returns>HydraCollection of NamedQuery</returns>
+    [HttpGet]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> GetNamedQueries(
+        [FromRoute] int customerId,
+        CancellationToken cancellationToken)
+    {
+        var namedQueries = new GetAllNamedQueries(customerId);
+        
+        return await HandleListFetch<DLCS.Model.Assets.NamedQueries.NamedQuery, GetAllNamedQueries, NamedQuery>(
+            namedQueries,
+            nq => nq.ToHydra(GetUrlRoots().BaseUrl),
+            errorTitle: "Failed to get Named Queries",
+            cancellationToken: cancellationToken
+        );
+    }
+    
+    /// <summary>
+    /// Create a new Named Query owned by the user - Only administrators may create a global named query
+    /// </summary>
+    /// <remarks>
+    /// Sample request:
+    ///
+    ///     POST: /customers/1/namedQueries
+    ///     {
+    ///         "name":"my-named-query",
+    ///         "template":"space=example"
+    ///     }
+    /// </remarks>
+    [HttpPost]
+    [ProducesResponseType(StatusCodes.Status201Created)]
+    [ProducesResponseType(StatusCodes.Status409Conflict)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    public async Task<IActionResult> PostNamedQuery(
+        [FromRoute] int customerId,
+        [FromBody] NamedQuery newNamedQuery,
+        [FromServices] HydraNamedQueryValidator validator,
+        CancellationToken cancellationToken)
+    {
+        if (newNamedQuery.Global == true && !User.IsAdmin()) 
+            return this.HydraProblem("Only admins are allowed to create global Named Queries", null, 400);
+        
+        var validationResult = await validator.ValidateAsync(newNamedQuery, cancellationToken);
+        if (!validationResult.IsValid)
+        {
+            return this.ValidationFailed(validationResult);
+        }
+        
+        var request = new CreateNamedQuery(customerId, newNamedQuery.ToDlcsModel(customerId));
+        
+        return await HandleUpsert(request, 
+            nq => nq.ToHydra(GetUrlRoots().BaseUrl),
+            errorTitle: "Failed to update Named Query",
```

Comment:
Should be `"Failed to create Named Query",`

Reply:
This has been changed in 4c918872ddb24d3158185bcd600fb9a1882d2a31